src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/ReferenceConfidenceModel.java

```diff
@@ -0,0 +1,547 @@
+     * confidence on the hom-ref call.
+     */
+    private static final byte BASE_QUAL_THRESHOLD = 6;
+
+    /**
+     * Only base calls with quality strictly greater than this constant,
+     * will be considered high quality if they are part of a soft-clip.
+     */
+    private static final byte HQ_BASE_QUALITY_SOFTCLIP_THRESHOLD = 28;
+
+    protected static final int MAX_N_INDEL_INFORMATIVE_READS = 40; // more than this is overkill because GQs are capped at 99 anyway
+    private static final int INITIAL_INDEL_LK_CACHE_PLOIDY_CAPACITY = 20;
+    private static GenotypeLikelihoods[][] indelPLCache = new GenotypeLikelihoods[INITIAL_INDEL_LK_CACHE_PLOIDY_CAPACITY + 1][];
+
+    /**
+     * Indel error rate for the indel model used to assess the confidence on the hom-ref call.
+     */
+    private static final double INDEL_ERROR_RATE = -4.5; // 10^-4.5 indel errors per bp
+
+    /**
+     * Phred scaled qual value that corresponds to the {@link #INDEL_ERROR_RATE indel error rate}.
+     */
+    private static final byte INDEL_QUAL = (byte) Math.round((INDEL_ERROR_RATE * -10.0));
+
+    /**
+     * No indel likelihood (ref allele) used in the indel model to assess the confidence on the hom-ref call.
+     */
+    private static final double NO_INDEL_LIKELIHOOD = QualityUtils.qualToProbLog10(INDEL_QUAL);
+
+    /**
+     * Indel likelihood (alt. allele) used in the indel model to assess the confidence on the hom-ref call.
+     */
+    private static final double INDEL_LIKELIHOOD = QualityUtils.qualToErrorProbLog10(INDEL_QUAL);
+
+    /**
+     * Create a new ReferenceConfidenceModel
+     *
+     * @param genomeLocParser how we create genome locs
+     * @param samples the list of all samples we'll be considering with this model
+     * @param header the SAMFileHeader describing the read information (used for debugging)
+     * @param indelInformativeDepthIndelSize the max size of indels to consider when calculating indel informative depths
+     */
+    public ReferenceConfidenceModel(final GenomeLocParser genomeLocParser,
+                                    final SampleList samples,
+                                    final SAMFileHeader header,
+                                    final int indelInformativeDepthIndelSize) {
+        Utils.nonNull(genomeLocParser, "genomeLocParser cannot be null");
+        Utils.nonNull(samples == null, "samples cannot be null");
+        if ( samples.numberOfSamples() == 0) throw new IllegalArgumentException("samples cannot be empty");
+        Utils.nonNull( header == null, "header cannot be empty");
+        if ( indelInformativeDepthIndelSize < 0) throw new IllegalArgumentException("indelInformativeDepthIndelSize must be >= 1 but got " + indelInformativeDepthIndelSize);
+
+        this.genomeLocParser = genomeLocParser;
+        this.samples = samples;
+        this.indelInformativeDepthIndelSize = indelInformativeDepthIndelSize;
+
+        if ( WRITE_DEBUGGING_BAM ) {
```

Comment:
Is this debugging something ongoing... like is useful to have it around or is just because there was a bug difficult to deal with that has been fixed? If the second I would remove this. Otherwise can you encapsulate this initialization into a private method?